Wobble/Graphics/UI/Form/Textbox.cs

```diff
@@ -439,6 +560,147 @@ public void ReadjustTextbox()
             FiredStoppedTypingActionHandlers = false;
         }
 
+        /// <summary>
+        ///    Makes sure the cursor is visible and resets its visiblity changing.
+        /// </summary>
+        public void ReadjustCursor()
+        {
+            Cursor.Visible = true;
+            TimeSinceCursorVisibllityChanged = 0;
+        }
+
+        /// <summary>
+        ///    Handles the arrow keys for the textbox.
+        /// </summary>
+        /// <param name="gameTime"></param>
+        private void HandleArrowKeys(GameTime gameTime)
+        {
+            if (!Focused)
+                return;
+
+            var shift = KeyboardManager.CurrentState.IsKeyDown(Keys.LeftShift) || KeyboardManager.CurrentState.IsKeyDown(Keys.RightShift);
+            var ctrl = KeyboardManager.CurrentState.IsKeyDown(Keys.LeftControl) || KeyboardManager.CurrentState.IsKeyDown(Keys.RightControl);
+
+            var upToCursor = RawText.Substring(0, CursorPosition);
+            var afterCursor = RawText.Substring(CursorPosition, RawText.Length - CursorPosition);
+
+            var oldCursorPosition = CursorPosition;
+
+            if (KeyboardManager.IsUniqueKeyPress(Keys.Left)
+            || (keyHeldFor.ContainsKey(Keys.Left) && keyHeldFor[Keys.Left] > 750
+                && gameTime.TotalGameTime.TotalMilliseconds - lastCursorMove > 75))
+            {
+                if (ctrl)
+                    MoveCursorToPrevious(c => char.IsWhiteSpace(c));
+                else
+                {
+                    if (string.IsNullOrEmpty(upToCursor) || char.IsLowSurrogate(upToCursor[upToCursor.Length - 1]))
+                    {
+                        CursorPosition = Math.Max(0, CursorPosition - 2);
+                    }
+                    else
+                    {
+                        CursorPosition = Math.Max(0, CursorPosition - 1);
+                    }
+                }
+
+                if (shift)
+                    SetSelectedPart(oldCursorPosition);
+
+                ReadjustCursor();
+                UpdateSelectedSprite();
+                lastCursorMove = gameTime.TotalGameTime.TotalMilliseconds;
+            }
+            if (KeyboardManager.IsUniqueKeyPress(Keys.Right)
+            || (keyHeldFor.ContainsKey(Keys.Right) && keyHeldFor[Keys.Right] > 750 
+                && gameTime.TotalGameTime.TotalMilliseconds - lastCursorMove > 75))
+            {
+                if (ctrl)
+                    MoveCursorToNext(c => char.IsWhiteSpace(c));
+                else
+                {
+                    if (string.IsNullOrEmpty(afterCursor) || char.IsHighSurrogate(afterCursor[0]))
+                    {
+                        CursorPosition = Math.Min(RawText.Length, CursorPosition + 2);
+                    }
+                    else
+                    {
+                        CursorPosition = Math.Min(RawText.Length, CursorPosition + 1);
+                    }
+                }
```

Comment:
Duplicate code. It would be worth putting this into its own method and reusing some of the logic.